hub/saml-soap-proxy/src/main/java/uk/gov/ida/hub/samlsoapproxy/healthcheck/HealthCheckData.java

```diff
@@ -0,0 +1,64 @@
+package uk.gov.ida.hub.samlsoapproxy.healthcheck;
+
+import com.google.common.base.Optional;
+
+class HealthCheckData {
+    
+    private static final String VERSION_DELIMITER = "version-";
+    private static final String EIDAS_DELIMITER = "eidasenabled-";
+    private static final String SHA_DELIMITER = "shouldsignwithsha1-";
+
+    private final String version;
+    private final String eidasEnabled;
+    private final String shouldSignWithSha1;
+    
+    public static HealthCheckData extractFrom(String responseId) {
```

Comment:
Hmm... can't we use pattern matcher. 

The substring and indexOf calls seems a bit... 1990s... Also bad things such as index out of bound can happen very easily...

Reply:
I agree (see commit message), and hoping to get something better in the pipeline once we have the msa working, since we can improve hub later.